src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -26,54 +26,155 @@
 #include <ripple/protocol/JsonFields.h>
 #include <ripple/protocol/TxFlags.h>
 #include <boost/format.hpp>
-#include <boost/tokenizer.hpp>
+#include <boost/regex.hpp>
 #include <algorithm>
 #include <mutex>
 
 namespace ripple {
+
+static
+std::vector<std::pair<uint256, std::string>>
+parseSection (Section const& section)
+{
+    static boost::regex const re1 (
+            "^"                       // start of line
+            "(?:\\s*)"                // whitespace (optional)
+            "([abcdefABCDEF0-9]{64})" // <hexadecimal amendment ID>
+            "(?:\\s+)"                // whitespace
+            "(\\S+)"                  // <description>
+        , boost::regex_constants::optimize
+    );
+
+    std::vector<std::pair<uint256, std::string>> names;
+
+    for (auto const& line : section.lines ())
+    {
+        boost::smatch match;
+
+        if (!boost::regex_match (line, match, re1))
+        {
+            throw std::runtime_error (
+                "Invalid entry in [" + section.name () + "]");
```

Comment:
Write out the line that we are trying to match so it's easy to find and correct in the config